elastic/kibana
# [Index Management] Improve error messages when editing index settings

Fixes https://github.com/elastic/kibana/issues/118026

### Summary
Currently when there's an error while saving index settings a very non informative error is shown to the user below the JSON editor. Which makes it harder for users to find if the viewport of their screen is narrow. This PR improves this error message by wrapping it in an error callout above the JSON editor and showing the actual error message returned by Elasticsearch instead of just a `Bad Request` label.

**How to test**
1. Start elasticsearch with `yarn es snapshot --license=trial` 
2. Start kibana with `yarn start`
3. Navigate to `Stack Management` -> `Index Management` -> open the flyout of any index and go to the `Edit settings` tab
4. Set `index.routing.allocation.include._tier_preference` to `"[data_content]"`
4. Verify that clicking the save button will show an error

**Screenshots**

<details>
<summary>before/after</summary>

// Old version
![Screenshot 2021-11-19 at 13 38 34](https://user-images.githubusercontent.com/1191206/142624128-3d31cfb3-bcd1-42ea-b720-fa43fb8327c9.png)

// Improved error handling
![Screenshot 2021-11-19 at 13 06 27](https://user-images.githubusercontent.com/1191206/142624484-38f61d86-3122-4174-84df-2e4f6731a4f4.png)
![Screenshot 2021-11-19 at 13 06 17](https://user-images.githubusercontent.com/1191206/142624478-119a5956-52df-4a80-be1b-4b9f34103d54.png)
</details>
